tests/angular-localForage.js

```diff
@@ -465,4 +477,94 @@ describe('Module: LocalForageModule', function() {
       }, done);
     });
   });
+
+  describe("createInstance", function () {
+      beforeEach(function () {
+          $localForage.createInstance({
+              name: 'DUPLICATE_INSTANCE_NAME'
+          });
+      });
+      it('should create a new instance', function () {
+          expect($localForage.createInstance({
+              name: 'TEST_INSTANCE'
+          })).toBeDefined();
+      });
+
+      it('should throw error if trying to create duplicate instance.',
+         function () {
+          expect($localForage.createInstance.bind($localForage, {
+              name: 'DUPLICATE_INSTANCE_NAME'
+          })).toThrowError(/already defined/);
+      });
+
+      it('should create instance with same name, different storeName',
+         function () {
+          expect($localForage.createInstance.bind($localForage, {
+              name: 'DUPLICATE_INSTANCE_NAME',
+              storeName: 'DIFFERENT_STORE_NAME'
+          })).not.toThrowError(/already defined/);
+      });
+  });
+
+  describe("instance", function () {
+      var $q, interval;
+      beforeEach(function () {
+          $localForage.createInstance({
+              name: 'TEST_INSTANCE_1'
+          });
+          $localForage.createInstance({
+              name: 'TEST_INSTANCE_2',
+              storeName: 'TEST_STORE_NAME_1'
+          });
+          $localForage.createInstance({
+              name: 'TEST_INSTANCE_2',
+              storeName: 'TEST_STORE_NAME_2'
+          });
+          inject(function (_$q_) {
+              $q = _$q_;
+          });
+          interval = triggerDigests();
+      });
+
+      afterEach(function () {
+          stopDigests(interval);
+      });
+
+      it('should get instance by name', function () {
+          expect($localForage.instance({
+              name: 'TEST_INSTANCE_1'
+          })).toBeDefined();
+      });
+
+      it('should throw exception if instance not exists', function () {
+          expect($localForage.instance.bind($localForage, {
+              name: 'NON_EXISTENT'
+          })).toThrowError();
+      });
+
+      it('should get instances with same name, different storeNames',
+         function (done) {
+          var instance1 = $localForage.instance({
+              name: 'TEST_INSTANCE_2',
+              storeName: 'TEST_STORE_NAME_1'
+          });
+          var instance2 = $localForage.instance({
+              name: 'TEST_INSTANCE_2',
+              storeName: 'TEST_STORE_NAME_2'
+          });
+          $q.all([
```

Comment:
Interesting. From my reading of the code you submitted, the second `var` would throw an error, is that right?

I would say as long as:
```js
var fooBar = { name: 'foo', storeName: 'bar' };
expect($localForage.instance(fooBar)).toBe($localForage.instance(fooBar));
```
is true, let's just simplify the test case to make it a little easier to understand what `instance` does in isolation from the other methods.

Might even be worth adding that as a test case, just to show that we're passing the instance by reference and expect that `instance` is (somewhat) referentially transparent?